### ISSUE TITLE
Any reference on equation (5) of the paper?

### ISSUE THREAD

**Issue by FirstHandScientist** [vgsatorras/hybrid-inference#2]
Hi,
Just be curious on the Equation (5) in your paper, which differs from maximum value of the hidden variable in a Hidden Markov model or in a linear dynamic model.
Is there any reference that defines the most likely state of x as a linear function of the derivative of log-likelihood?

Additionally, in the loss function defined in (18), gt seems to be the ground truth of GNN variable h, which does not seem to be available since there would only exist groud truth of x and y.

Your explanation would be appreciated. Thanks.

**Comment by vgsatorras** [comment]
Hi,

Thank you for the interest in the paper.
Equation (5) is an optimization iterative algorithm (gradient descent) on the negative log probability. Gradient descent is guaranteed to converge to the most likely state of x when the negative log-likelihood function is convex (e.g. gaussian case).  Another work using the same procedure would be [Recurrent Inference Machines for Solving Inverse Problems](https://arxiv.org/pdf/1706.04008.pdf).

In the loss function defined in (18), gt is the ground truth for the totality or part of the variable x. In some experiments, for example, the GPS experiment, x represents (position, velocity and acceleration) but we may only have access to the ground truth of the position, then we used a function selector \phi(x), that outputs the position which will be used to compute the loss with the gt.
